slack_integration: Avoid showing senders email.

PieterCK opened the pull request:
Fixes: https://github.com/zulip/zulip/issues/36090

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**
Before:
<img width="1520" height="200" alt="image" src="https://github.com/user-attachments/assets/c9b86e12-14dd-4aea-b791-40194ea5d13f" />

After:
<img width="1416" height="232" alt="image" src="https://github.com/user-attachments/assets/00df3dd7-3e57-41df-a086-97fafdd81561" />

<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [ ] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [ ] Explains differences from previous plans (e.g., issue description).
- [ ] Highlights technical choices and bugs encountered.
- [ ] Calls out remaining decisions and concerns.
- [ ] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [ ] Each commit is a coherent idea.
- [ ] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [ ] Visual appearance of the changes.
- [ ] Responsiveness and internationalization.
- [ ] Strings and tooltips.
- [ ] End-to-end functionality of buttons, interactions and flows.
- [ ] Corner cases, error conditions, and easily imagined bugs.
</details>

PieterCK commented on zerver/webhooks/slack/fixtures/slack_conversations_info_api_response.json:
https://docs.slack.dev/reference/methods/conversations.info/

PieterCK commented:
Updated the notification screenshot in the doc

alya commented:
Thanks! @Niloth-p and/or @laurynmm , could you please review? I'd like to get this one merged quickly if we can.

PieterCK commented:
@Niloth-p Thanks for checking this out! Changes:
- Fixed `"Salck"` -> `"Slack"` string in a commit.
- And removed the Slack integration from the automated screenshot tool in this commit: `integrations: Remove Slack from WEBHOOK_SCRENNSHOT_CONFIG.`

PieterCK commented:
Ahh okay, I'll drop that commit then

JonathanLindsey commented:
My [comment](https://github.com/zulip/zulip/issues/36090#issuecomment-3368352229) on the issue:
> [@PieterCK](https://github.com/PieterCK) I got a development environment setup and tried [#36093](https://github.com/zulip/zulip/pull/36093) without giving the Slack bot the `users:read.email` scope. It didn't work. But I then removed most of the instances of `users:read.email` from the codebase (all of them except for documentation) which you can see in [my own branch](https://github.com/JonathanLindsey/zulip/tree/fix-no-email-slack-use-display-name). This worked. Of course it needs polish to pass tests and everything else, but I believe this shows that the `users:read.email` scope should be unnecessary.

Next I tried this PR without re-setting up the Slack integration and was able to pass a message from Slack into zulip. However, when I change the Event Subscription Request URL on the Slack integration, it tries to re-verify and fails. I'm guessing you can remove some checks for `users:read.email` and we won't require that scope anymore.

PieterCK commented:
Hey, I rechecked the toke scopes again and yeah it looks like `users.read.email` and a couple other scopes are not required for the integration. I didn't get any error when trying to register the webhook URL to Slack, though. If you're using a dev environment to test things out, you'll have to patch a couple of validations.

I've opened #36196 to fix the required scopes. That PR and this PR can be merged separately.

timabbott commented on zerver/webhooks/slack/tests.py:
```diff
@@ -1,53 +1,54 @@
 
 MESSAGE_WITH_NORMAL_TEXT = "Hello, this is a normal text message"
-USER = "John Doe"
+USER = "supersecretemail"
```
OK I read https://docs.slack.dev/reference/methods/users.info/, and `name` is their Slack username, which like `tabbott` would often be the prefix of an email address, but it's typically a free choice; like I'm sure there are orgs where one might have an email of `tim@example.com` but a username of `tabbott` or whatever that person likes.

So we should not be calling this an email in the commit message and PR description, we should be calling it their "Slack username". I agree it's more in line with Zulip patterns to use a full name rather than a Slack username, so the change is likely a good one, but we should describe it accurately.

Also there's a typo in the first commit's commit message.

PieterCK replied:
Ok, updated the last commit description to:
```
slack_integration: Avoid showing sender username.

The "name" field included in the users.info API response is actually the
senders Slack username, which may be the prefix of their email address.

This updates the integration to use the "real_name" field instead, which
is the senders full name.

Fixes https://github.com/zulip/zulip/issues/36090.
```

And fixed the typo in the first commit: `WEBHOOK_SCRENNSHOT_CONFIG` → `WEBHOOK_SCREENSHOT_CONFIG`

PieterCK commented:
This is ready for review again
